src/bin/client-demo.rs

```diff
@@ -13,22 +13,25 @@ use std::thread::sleep;
 use std::time::{Duration, Instant};
 
 fn main() {
-    let addr = "127.0.0.1:8000";
-    let send_addr = "127.0.0.1:8001";
+    let addr = "127.0.0.1:9000";
+    let send_addr = "127.0.0.1:9001";
```

Comment:
Not sure what I was thinking with that name `send_addr`. Can you rename that to `reply_addr` or `response_addr`? Also, that port can probably be `0`, eh?